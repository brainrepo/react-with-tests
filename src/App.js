import "./App.css";
import Sidebar from "./containers/Sidebar";
import Configurations from "./containers/Configurations";
import { Switch, Route } from "react-router-dom";

function App() {
  return (
    <>
      <Sidebar />
      <Switch>
        <Route path="/configurations">
          <Configurations />
        </Route>
        <Route path="/">
          <div>to replace</div>
        </Route>
      </Switch>
    </>
  );
}

export default App