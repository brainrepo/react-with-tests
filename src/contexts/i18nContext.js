import {createContext, useContext, useState} from 'react';

import it from "../assets/i18n/it.json";
import en from "../assets/i18n/en.json";
import fr from "../assets/i18n/fr.json";

const langs = {en, fr, it}

const I18nStateContext = createContext();
const I18nSetContext = createContext();

function I18nContextProvider({children}) {
    const [lang, setLang] = useState("en");
    return (
      <I18nStateContext.Provider value={lang}>
          <I18nSetContext.Provider value={setLang}>
              {children}
          </I18nSetContext.Provider>
      </I18nStateContext.Provider>
    )
}

function useI18nState() {
    const context = useContext(I18nStateContext);

    if (context === undefined) {
        throw new Error("useI18nState must be used inside a I18nContext provider")
    }

    return langs[context];
}

function useI18nSet() {
    const context = useContext(I18nSetContext);

    if (context === undefined) {
        throw new Error("useI18nState must be used inside a I18nContext provider")
    }

    return context;
}

function useI18n(){
    return [useI18nState(), useI18nSet()]
}

export {I18nContextProvider, useI18nState, useI18nSet, useI18n}
