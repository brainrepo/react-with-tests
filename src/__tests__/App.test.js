import {
  fireEvent,
  render,
  screen,
  waitFor,
  cleanup,
} from "@testing-library/react";
import App from "../App";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import {I18nContextProvider} from "../contexts/i18nContext"

let renderResult = null;
let history = null;

beforeEach(() => {
  history = createMemoryHistory();
  renderResult = render(
    <I18nContextProvider>
      <Router history={history}>
        <App />
      </Router>
    </I18nContextProvider>
  );
});

afterEach(cleanup);

describe("<App/>", () => {
  it("can match the snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });
  it("can go to configuration area", async () => {
    const { getByText } = screen;
    const el = getByText(/Configurations/);
    fireEvent.click(el);
    await waitFor(() => getByText("Configuration area"));
    expect(getByText("Configuration area")).toBeVisible();
  });
});
