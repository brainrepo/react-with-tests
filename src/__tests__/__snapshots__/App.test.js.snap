// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> can match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="header"
      >
        <svg
          fill="none"
          height="47"
          viewBox="0 0 48 47"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0)"
          >
            <path
              d="M47.5 0H0.5V47H47.5V0Z"
              fill="white"
            />
            <path
              d="M30.1066 43.857L17.9227 43.9317V40.6304L17.724 40.5687V39.9669H18.1874C18.2433 39.824 18.2979 39.6817 18.3517 39.5388L17.724 39.3425V38.7407H18.644C19.5337 36.2434 20.2341 33.6019 20.7843 30.9338L20.3627 30.7073V30.1055H20.9493C23.4485 17.2694 22.5939 10.04 22.7859 9.70674V9.38897H22.5394V8.88738H22.7859V8.72016H22.5394V8.21857H22.8034L23.5833 7.44956H23.9342V6.95385H23.6816V6.70839H23.9342V6.48901H23.6816V6.24473H23.9342V6.02594H23.6816V5.78047H23.9342V5.56109H23.6816V5.31563H23.9342V3.07618C23.9342 3.03349 23.9745 2.9985 24.0231 2.9985C24.0715 2.9985 24.1119 3.0329 24.1119 3.07618V5.31563H24.3652V5.5605H24.1119V5.77988H24.3652V6.02475H24.1119V6.24413H24.3652V6.48901H24.1119V6.70839H24.3652V6.95208H24.1119V7.44774H24.3807L25.1612 8.21676H25.4252V8.71835H25.1788V8.88557H25.4252V9.38716H25.1788V9.705C25.37 10.0388 24.5349 17.2677 27.0463 30.1037H27.6336V30.7945L27.2268 30.9966C27.7783 33.6428 28.4774 36.2617 29.361 38.7389H30.2062V39.4297L29.6721 39.5844C29.72 39.7114 29.7691 39.8382 29.8183 39.9651H30.2062V40.6553L30.1066 40.6837V43.857ZM21.4073 38.7389H26.5715C26.3439 38.605 25.3565 32.9563 25.3565 32.9563H22.5845C22.5845 34.2269 21.4073 38.7389 21.4073 38.7389Z"
              fill="#1A202C"
            />
            <path
              d="M5.52031 22.4979V21.6222C5.52031 21.3024 5.72186 21.0105 6.12496 20.7465L14.4651 15.1169C14.5345 15.0752 14.6179 15.0544 14.7153 15.0544C14.8264 15.0544 14.9238 15.0961 15.0072 15.1795C15.0906 15.2629 15.1323 15.3671 15.1323 15.4922V18.87C15.1323 19.2174 14.9098 19.5233 14.4651 19.7874L10.8163 22.06L14.4651 24.3535C14.604 24.4369 14.7013 24.4995 14.757 24.5412C14.8264 24.5829 14.9029 24.6733 14.9863 24.8122C15.0836 24.9373 15.1323 25.0833 15.1323 25.2501V28.6278C15.1323 28.7529 15.0906 28.8572 15.0072 28.9406C14.9238 29.024 14.8264 29.0657 14.7153 29.0657C14.6179 29.0657 14.5345 29.0448 14.4651 29.0031L6.12496 23.3736C5.72186 23.1095 5.52031 22.8176 5.52031 22.4979Z"
              fill="black"
            />
            <path
              d="M42.4744 21.6222V22.4979C42.4744 22.8176 42.2729 23.1095 41.8698 23.3736L33.5297 29.0031C33.4603 29.0448 33.3699 29.0657 33.2587 29.0657C33.1614 29.0657 33.071 29.024 32.9876 28.9406C32.9042 28.8572 32.8625 28.7529 32.8625 28.6278V25.2501C32.8625 25.0833 32.9042 24.9373 32.9876 24.8122C33.0849 24.6733 33.1614 24.5829 33.217 24.5412C33.2865 24.4995 33.3907 24.4369 33.5297 24.3535L37.1785 22.06L33.5297 19.7874C33.085 19.5233 32.8625 19.2174 32.8625 18.87V15.4922C32.8625 15.3671 32.9042 15.2629 32.9876 15.1795C33.071 15.0961 33.1614 15.0544 33.2587 15.0544C33.3699 15.0544 33.4603 15.0752 33.5297 15.1169L41.8698 20.7465C42.2729 21.0105 42.4744 21.3024 42.4744 21.6222Z"
              fill="black"
            />
          </g>
          <defs>
            <clippath
              id="clip0"
            >
              <rect
                fill="white"
                height="47"
                transform="translate(0.5)"
                width="47"
              />
            </clippath>
          </defs>
        </svg>
      </div>
      <ul
        class="links"
      >
        <li>
          <a
            alt=""
            aria-current="page"
            class="active"
            href="/"
          >
            Podcasts
          </a>
        </li>
        <li>
          <a
            alt=""
            href="/configurations"
          >
            Configurations
          </a>
        </li>
      </ul>
    </div>
    <div>
      to replace
    </div>
  </div>
</DocumentFragment>
`;
